CI: build-windows-msys2-mingw disabled

An upgrade in the msys2-mingw toolchain causes lots of warnings and errors to appear: https://github.com/OSGeo/gdal/runs/5216450145?check_suite_focus=true
We can not differences in the following packages:
``
$ diff -u /tmp/old.txt /tmp/new.txt
--- /tmp/old.txt	2022-02-16 14:56:34.422230084 +0100
+++ /tmp/new.txt	2022-02-16 14:56:22.422319897 +0100
@@ -1,4 +1,3 @@
-: Retrieving packages...
 mingw-w64-x86_64-gcc-11.2.0-8-any downloading...
 mingw-w64-x86_64-gcc-ada-11.2.0-8-any downloading...
 mingw-w64-x86_64-python-3.9.10-2-any downloading...
@@ -8,11 +7,11 @@
 mingw-w64-x86_64-cmake-3.22.2-1-any downloading...
 mingw-w64-x86_64-sqlite3-3.37.2-1-any downloading...
 mingw-w64-x86_64-gdb-multiarch-11.2-2-any downloading...
-mingw-w64-x86_64-binutils-2.37-4-any downloading...
-mingw-w64-x86_64-headers-git-9.0.0.6373.5be8fcd83-1-any downloading...
+mingw-w64-x86_64-binutils-2.37-5-any downloading...
+mingw-w64-x86_64-headers-git-9.0.0.6373.5be8fcd83-2-any downloading...
 mingw-w64-x86_64-openssl-1.1.1.m-1-any downloading...
 mingw-w64-x86_64-gdb-11.2-2-any downloading...
-mingw-w64-x86_64-crt-git-9.0.0.6373.5be8fcd83-1-any downloading...
+mingw-w64-x86_64-crt-git-9.0.0.6373.5be8fcd83-2-any downloading...
 mingw-w64-x86_64-gettext-0.21-3-any downloading...
 mingw-w64-x86_64-tcl-8.6.11-5-any downloading...
 mingw-w64-x86_64-tk-8.6.11.1-2-any downloading...
@@ -60,11 +59,11 @@
 mingw-w64-x86_64-mpc-1.2.1-1-any downloading...
 mingw-w64-x86_64-jansson-2.14-1-any downloading...
 mingw-w64-x86_64-libffi-3.3-4-any downloading...
-mingw-w64-x86_64-winpthreads-git-9.0.0.6373.5be8fcd83-1-any downloading...
-mingw-w64-x86_64-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any downloading...
+mingw-w64-x86_64-winpthreads-git-9.0.0.6373.5be8fcd83-2-any downloading...
+mingw-w64-x86_64-libwinpthread-git-9.0.0.6373.5be8fcd83-2-any downloading...
 mingw-w64-x86_64-termcap-1.3.1-6-any downloading...
 mingw-w64-x86_64-libsystre-1.0.1-4-any downloading...
 mingw-w64-x86_64-libb2-0.98.1-1-any downloading...
 mingw-w64-x86_64-libmangle-git-9.0.0.6373.5be8fcd83-1-any downloading...
-mingw-w64-x86_64-winstorecompat-git-9.0.0.6373.5be8fcd83-1-any downloading...
-mingw-w64-x86_64-windows-default-manifest-6.4-3-any downloading...
+mingw-w64-x86_64-winstorecompat-git-9.0.0.6373.5be8fcd83-2-any downloading...
+mingw-w64-x86_64-windows-default-manifest-6.4-4-any downloading...

@dg0yt: Looks like it is mixing a different (older) compiler with the headers from MSYS2/MINGW64.
During configure:
```
-- The C compiler identification is GNU 8.1.0
-- The CXX compiler identification is GNU 8.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/ProgramData/chocolatey/bin/gcc.exe - skipped
```
When building:
```
C:/tools/msys64/mingw64/include/stdio.h:1249:29: error: use of old-style cast to 'wint_t' {aka 'short unsigned int'} [-Werror=old-style-cast]
```

GCC in MSYS2/MINGW64 is currently version 11.2.0.

Does it work for you if you are using the compilers from MSYS2/MINGW64?

--- a/.github/workflows/cmake_builds.yml
+++ b/.github/workflows/cmake_builds.yml
@@ -131,7 +131,6 @@ jobs:
         fi
 
   build-windows-msys2-mingw:
-    if: ${{ false }}  # disable for now. See https://github.com/OSGeo/gdal/issues/5317
     runs-on: windows-2022
     env:
       generator: MinGW Makefiles
@@ -175,8 +174,9 @@ jobs:
       #  libxml2 disabled because currently causes a 'Imported target "LibXml2::LibXml2" includes non-existent path "/mingw64/include/libxml2" in its INTERFACE_INCLUDE_DIRECTORIES'
       # Disable mySQL since C:/mysql/lib/mysqlclient.lib (unrelated to msys) is found, which causes linking issues
       # Disable Python bindings because of 'ValueError: filename D:/a/gdal/gdal/swig/python/osgeo/__init__.py does not start with the input_base_dir D:/a/gdal/gdal/swig/python/osgeo/\' when running lib2to3
+      # Set explicitly CMAKE_C|CXX_COMPILER otherwise C:/ProgramData/chocolatey/bin/gcc.exe would be used
       - name: Configure
-        run: cmake -DCMAKE_BUILD_TYPE=release -G "${env:generator}" -Werror=dev "-DCMAKE_C_COMPILER_LAUNCHER=c:\tools\msys64\mingw64\bin\ccache.exe" "-DCMAKE_CXX_COMPILER_LAUNCHER=c:\tools\msys64\mingw64\bin\ccache.exe" "-DCMAKE_PREFIX_PATH=C:\tools\msys64\mingw64" "-DCMAKE_UNITY_BUILD=${env:CMAKE_UNITY_BUILD}" -S . -B "build" -DLIBXML2_INCLUDE_DIR=C:/tools/msys64/mingw64/include -DGDAL_USE_LIBXML2:BOOL=OFF -DGDAL_USE_MYSQL:BOOL=OFF -DBUILD_PYTHON_BINDINGS:BOOL=OFF -DBUILD_CSHARP_BINDINGS=ON -DCMAKE_C_FLAGS=-Werror -DCMAKE_CXX_FLAGS=-Werror -DWERROR_DEV_FLAG="-Werror=dev"
+        run: cmake -DCMAKE_BUILD_TYPE=release -G "${env:generator}" -Werror=dev "-DCMAKE_C_COMPILER=c:/tools/msys64/mingw64/bin/gcc.exe" "-DCMAKE_CXX_COMPILER=c:/tools/msys64/mingw64/bin/g++.exe" "-DCMAKE_C_COMPILER_LAUNCHER=c:\tools\msys64\mingw64\bin\ccache.exe" "-DCMAKE_CXX_COMPILER_LAUNCHER=c:\tools\msys64\mingw64\bin\ccache.exe" "-DCMAKE_PREFIX_PATH=C:\tools\msys64\mingw64" "-DCMAKE_UNITY_BUILD=${env:CMAKE_UNITY_BUILD}" -S . -B "build" -DLIBXML2_INCLUDE_DIR=C:/tools/msys64/mingw64/include -DGDAL_USE_LIBXML2:BOOL=OFF -DGDAL_USE_MYSQL:BOOL=OFF -DBUILD_PYTHON_BINDINGS:BOOL=OFF -DBUILD_CSHARP_BINDINGS=ON -DCMAKE_C_FLAGS=-Werror -DCMAKE_CXX_FLAGS=-Werror -DWERROR_DEV_FLAG="-Werror=dev"
         working-directory: ${{ github.workspace }}
       - name: Build
         run: cmake --build build -j 3

--- a/frmts/ceos2/ceos.c
+++ b/frmts/ceos2/ceos.c
@@ -222,7 +222,7 @@ void GetCeosField(CeosRecord_t *record, int32 start_byte,
     HFree(mod_buf);
 }
 
-void SetCeosField(CeosRecord_t *record, int32 start_byte, char *format, void *value)
+void SetCeosField(CeosRecord_t *record, int32 start_byte, const char *format, int intValue, double dblValue)
 {
     int field_size;
     char * temp_buf = NULL;
@@ -250,6 +250,7 @@ void SetCeosField(CeosRecord_t *record, int32 start_byte, char *format, void *va
     {
     case 'b':
     case 'B':
+#if 0
         /* Binary data type */
         if(field_size > 1)
         {
@@ -258,33 +259,44 @@ void SetCeosField(CeosRecord_t *record, int32 start_byte, char *format, void *va
             memcpy(value,temp_buf,field_size);
         }
         break;
+#endif
+        fprintf(stderr, "SetCeosField with format=%c not implemented",
+                format[0]);
+        HFree(temp_buf);
+        return;
 
     case 'i':
     case 'I':
         /* Integer data type */
         snprintf( szPrintfFormat, sizeof(szPrintfFormat), "%%%s%c",format+1, 'd');
-        snprintf( temp_buf, field_size+1, szPrintfFormat, *(int *) value);
+        snprintf( temp_buf, field_size+1, szPrintfFormat, intValue);
         break;
 
     case 'f':
     case 'F':
         /* Double precision floating point data type */
         snprintf( szPrintfFormat, sizeof(szPrintfFormat), "%%%s%c", format+1, 'g');
-        snprintf( temp_buf, field_size+1, szPrintfFormat, *(double *)value);
+        snprintf( temp_buf, field_size+1, szPrintfFormat, dblValue);
         break;
 
     case 'e':
     case 'E':
         /* Double precision floating point data type (forced exponent) */
         snprintf( szPrintfFormat, sizeof(szPrintfFormat), "%%%s%c", format+1, 'e');
-        snprintf( temp_buf, field_size+1, szPrintfFormat, *(double *)value);
+        snprintf( temp_buf, field_size+1, szPrintfFormat, dblValue);
         break;
 
     case 'a':
     case 'A':
+#if 0
         strncpy(temp_buf,value,field_size+1);
         temp_buf[field_size] = '0';
         break;
+#endif
+        fprintf(stderr, "SetCeosField with format=%c not implemented",
+                format[0]);
+        HFree(temp_buf);
+        return;
 
     default:
         /* Unknown format */
@@ -303,7 +315,7 @@ void SetIntCeosField(CeosRecord_t *record, int32 start_byte, int32 length, int32
     char total_len[12];   /* 12 because 2^32 -> 4294967296 + I + null */
 
     snprintf(total_len,sizeof(total_len),"I%d",length);
-    SetCeosField(record,start_byte,total_len,&integer_value);
+    SetCeosField(record,start_byte,total_len,integer_value,0.0);
 }
 
 CeosRecord_t *FindCeosRecord(Link_t *record_list, CeosTypeCode_t typecode, int32 fileid, int32 flavor, int32 subsequence)

--- a/frmts/ceos2/ceos.h
+++ b/frmts/ceos2/ceos.h
@@ -275,7 +275,7 @@ void PutCeosRecordStruct(CeosRecord_t *record, const void *struct_ptr);
 
 void GetCeosField(CeosRecord_t *, int32, const char *, void *);
 
-void SetCeosField(CeosRecord_t *record, int32 start_byte, char *format, void *value);
+void SetCeosField(CeosRecord_t *record, int32 start_byte, const char *format, int intValue, double dblValue);
 
 void SetIntCeosField(CeosRecord_t *record, int32 start_byte, int32 length, int32 value);
 

--- a/ogr/ogrsf_frmts/mssqlspatial/include_msodbcsql.h
+++ b/ogr/ogrsf_frmts/mssqlspatial/include_msodbcsql.h
@@ -1,0 +1,47 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  Includes msodbcsql.h
+ * Author:   Even Rouault <even dot rouault at spatialys dot com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2022, Even Rouault <even dot rouault at spatialys dot com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *****************************************************************************/
+
+#ifndef INCLUDE_MSODBCSQL_H_DEFINED
+#define INCLUDE_MSODBCSQL_H_DEFINED
+
+#if defined(__GNUC__)
+#pragma GCC system_header
+
+// msodbcsql.h uses Microsoft SAL annotations which aren't set by mingw64
+#ifndef __in
+#define __in
+#define __in_z
+#define __out
+#define __out_ecount_z_opt(x)
+#endif
+
+#endif
+
+#include <msodbcsql.h>
+
+#endif

--- a/ogr/ogrsf_frmts/mssqlspatial/ogr_mssqlspatial.h
+++ b/ogr/ogrsf_frmts/mssqlspatial/ogr_mssqlspatial.h
@@ -38,7 +38,7 @@
 #include <sqlncli.h>
 #endif
 #ifdef MSODBCSQL_VERSION
-#include <msodbcsql.h>
+#include "include_msodbcsql.h"
 #endif
 
 class OGRMSSQLSpatialDataSource;

